### DISCUSSION TITLE
How to use raw DB queries as data for a screen?

### DISCUSSION THREAD

**Discussion by jakoubek** [orchidsoftware/platform#1769] [Q&A] [answered]
Hi,

what is the proposed way to encapsulate the result of a raw database query (one that does NOT return an Eloquent collection, but instead an array of arrays) into a Repository?

This does not work:

```php
return [
            'accounts' => DB::connection('mydbconnection')
                            ->select('SELECT id, name FROM accounts'),
        ];
```

... since it leads to that understandable error:

```
Call to a member function getContent() on array
```

Do I have to iterate over every array element and put it into an Repository?

Thanks,
Oli

**Comment by tabuna** [comment, marked as answer]
Hi @jakoubek. The short answer is yes:

```php
$result = DB::connection('mydbconnection')->select('SELECT id, name FROM accounts');
 
$accounts = collect($result)->transform(fn(array $account) => new \Orchid\Screen\Repository($account));
```

You may be wondering why is this needed? This is necessary to be consistent. 
Since in the columns, we allow you to refer to the depth of the object using dot notation, for example:

```php
TD::make('accounts.balance');
```

And other similar things. Therefore, we require that it be defined, either in the class of the array was wrapped in an object that will allow you to do this.

**Reply by jakoubek** [reply]
Thanks @tabuna ,

works fine - thank you!
